### PR TITLE
BoxView: tests cases

### PR DESCRIPTION
**Pull request by vahidbazzaz**
The PR contains test cases for more important BoxView tests based on updated test planes

### CODE REVIEW

**Comment by vahidbazzaz** [comment]
[2.0 BoxView: write tests](https://trello.com/c/DZF1ksfp)

**Review by HaGuesto — APPROVED** [review]
generally you can merge to keep moving forward.

Some general feedback:
- check with Daniel if mocking data can be done easier with msw
- in the response mocks, do not include locations like LOST and SCRAP
- always import the graphql queries/mutations
- should we split up tests for subcomponents? for sure when they become global components
- extract BoxView graphql into its own file

**Comment by pylipp** [comment]
@vahidbazzaz can you create and merge a quick PR with the formatting applied to the `src/views/Box` directory, and then merge master into this branch such that reviewing becomes clearer?

File: react/src/views/Box/BoxView.test.tsx [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+        fireEvent.click(screen.getByText("Submit"));
+
+        await waitFor(() => {
+            expect(screen.getByTestId("boxview-number-items")).toBeInTheDocument();
+            expect(screen.getByTestId("boxview-number-items")).toHaveTextContent(
+                "# " + numberOfItemWhenIncreased,
+            );
+        });
+
+        // fireEvent.click(screen.getByTestId("decrease-items"));
```
Comment (pylipp) [inline react/src/views/Box/BoxView.test.tsx:365]: Do you plan to still implement this test?

File: react/src/views/Box/BoxView.test.tsx [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+        // await waitFor(() => userEvent.type(screen.getByTestId("decrease-number-items"), "1"));
+
+        // await waitFor(() => {
+        //     expect(screen.getByTestId("boxview-number-items")).toBeInTheDocument();
+        //     expect(screen.getByTestId("boxview-number-items")).toHaveTextContent(
+        //         "# " + numberOfItemWhenDecreased,
+        //     );
+        // });
+    });
+    // Test case 3.1.1.4
```
Comment (pylipp) [inline react/src/views/Box/BoxView.test.tsx:375]: Also 3.1.1.4.1 is included in the test implementation already

File: react/src/views/Box/BoxView.test.tsx [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+            const boxSubheading = screen.getByTestId("box-subheader");
+            expect(boxSubheading).toHaveTextContent(`State: Scrap`);
+            expect(screen.getByTestId("box-state")).toHaveStyle(`color: #EB404A`);
+        });
+    });
+    // Test case 3.1.1.5
+    it("3.1.1.5 - click on a location should move box to seleted place", async () => {
+        await waitFor(waitTillLoadingIsDone);
+
+        // fireEvent.click(screen.getByTestId("location-wh1-btn"));
```
Comment (pylipp) [inline react/src/views/Box/BoxView.test.tsx:397]: still to be implemented?

File: react/src/views/Box/BoxView.test.tsx [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+        //     const boxLocationLabel = screen.getByTestId("box-location-label");
+        //     expect(boxLocationLabel).toHaveTextContent(`Move this box from WH1 to:`);
+        // });
+    });
+
+    // Test case 3.1.1.6
+    it("3.1.1.6 - If Distro Event Not Available it should not shown the ui section for that", async () => {
+        await waitFor(waitTillLoadingIsDone);
+        const distroEventSection = screen.getByTestId("distro-event-section");
+        expect(distroEventSection).toBeInTheDocument();
```
Comment (pylipp) [inline react/src/views/Box/BoxView.test.tsx:408]: How does this test verify that no distro event is shown?

File: react/src/views/Box/BoxView.test.tsx [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+    });
+    // Test case 3.1.1.7
+    it("3.1.1.7 - render tags correctly if box assigned tags", async () => {
+        await waitFor(waitTillLoadingIsDone);
+        const boxTags = screen.getByTestId("box-tags");
+        expect(boxTags).toBeInTheDocument();
```
Comment (pylipp) [inline react/src/views/Box/BoxView.test.tsx:414]: I noticed there are `tags` data in the mocks, is it hard to verify that the tag is actually shown in the UI?

File: react/src/views/Box/BoxView.test.tsx (hunk 1 above)
Comment (vahidbazzaz) [reply react/src/views/Box/BoxView.test.tsx:365]: @pylipp it make mocking date quiet complex so I skip this for time being

File: react/src/views/Box/BoxView.test.tsx (hunk 3 above)
Comment (vahidbazzaz) [reply react/src/views/Box/BoxView.test.tsx:397]: @pylipp 

> I could understand and follow most tests :) do you intend to implement the other tests from the 3.1.1 section in this PR or in another one?

@pylipp I think it make sense to skip other tests for another PR and also as we discussed we just implement few important one :)

File: react/src/views/Box/BoxView.test.tsx (hunk 5 above)
Comment (vahidbazzaz) [reply react/src/views/Box/BoxView.test.tsx:414]: @pylipp I will try to verify `tags` per your suggestion

**Review by pylipp — APPROVED** [review]

**Pull request merged**